hbase/src/main/java/org/apache/zeppelin/hbase/HbaseInterpreter.java

```diff
@@ -0,0 +1,152 @@
+ * hbase.ruby.sources: Dir where shell ruby code is installed.
+ *                          Path is relative to hbase.home. Default: lib/ruby
+ * hbase.irb.load: (Testing only) Default is true.
+ *                      Whether to load irb in the interpreter.
+ */
+public class HbaseInterpreter extends Interpreter {
+  private Logger logger = LoggerFactory.getLogger(HbaseInterpreter.class);
+  private ScriptingContainer scriptingContainer;
+
+  private StringWriter writer;
+
+  static {
+    Interpreter.register("hbase", "hbase", HbaseInterpreter.class.getName(),
+        new InterpreterPropertyBuilder()
+            .add("hbase.home", "/usr/lib/hbase/", "Installation dir. of Hbase")
+            .add("hbase.ruby.sources", "lib/ruby",
+                "Path to Ruby scripts relative to 'hbase.home'")
+            .add("hbase.irb.load", "true", "Load hirb. hirb loads hbase shell. " +
+                "For unit tests it has to be turned off.")
+            .build());
+  }
+
+  public HbaseInterpreter(Properties property) {
+    super(property);
+  }
+
+  @Override
+  public void open() {
+    String hbase_home = getProperty("hbase.home");
+    String ruby_src = getProperty("hbase.ruby.sources");
+    String abs_ruby_src = hbase_home + ruby_src;
+
+    File f = new File(abs_ruby_src);
+    if (!f.exists() || !f.isDirectory()) {
+      throw new InterpreterException("hbase ruby sources is not correctly defined.");
+    }
+
+    logger.info("Home:" + hbase_home);
+    logger.info("Ruby Src:" + ruby_src);
+
+    Properties props = System.getProperties();
+    props.setProperty("hbase.ruby.sources", abs_ruby_src);
+    this.scriptingContainer  = new ScriptingContainer(LocalContextScope.SINGLETON);
+    List<String> paths = new ArrayList<>(Arrays.asList(abs_ruby_src));
+    this.writer = new StringWriter();
+    scriptingContainer.setOutput(this.writer);
+    this.scriptingContainer.setLoadPaths(paths);
+    scriptingContainer.setCompatVersion(org.jruby.CompatVersion.RUBY1_9);
+    if (Boolean.parseBoolean(getProperty("hbase.irb.load"))) {
+      try {
+        InputStream in = getClass().getResourceAsStream("/hbase/bin/hirb.rb");
+        scriptingContainer.runScriptlet(in, "/hbase/bin/hirb.rb");
+        in.close();
+      } catch (NullPointerException | IOException e) {
+        logger.error("Open failed:", e);
+      }
+    }
+  }
+
+  @Override
+  public void close() {
+    if (this.scriptingContainer != null) {
+      this.scriptingContainer.terminate();
+    }
+  }
+
+  @Override
+  public InterpreterResult interpret(String cmd, InterpreterContext interpreterContext) {
+    try {
+      logger.info(cmd);
+      this.writer.getBuffer().setLength(0);
+      this.scriptingContainer.runScriptlet(cmd);
+      this.writer.flush();
+      logger.debug(writer.toString());
+      return new InterpreterResult(InterpreterResult.Code.SUCCESS, writer.getBuffer().toString());
+    } catch (Throwable t) {
+      logger.error("Can not run " + cmd, t);
```

Comment:
cmd can be a long string? or multiple lines? suggest putting this in quote for clarity